src/cursor/cur_hs.c

```diff
@@ -938,10 +939,11 @@ __curhs_insert(WT_CURSOR *cursor)
 
 #ifdef HAVE_DIAGNOSTIC
     /* Do a search again and call next to check the key order. */
-    WT_WITH_PAGE_INDEX(session, ret = __curhs_search(cbt, false));
-    WT_ASSERT(session, ret == 0);
-    if (cbt->compare == 0)
+    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = file_cursor->search(file_cursor));
+    if (ret == 0)
         WT_ERR_NOTFOUND_OK(__curhs_file_cursor_next(session, file_cursor), false);
+    else
+        WT_ASSERT(session, ret == WT_NOTFOUND);
```

Comment:
Why do you expect the search can return WT_NOTFOUND? The update is just inserted.

Reply:
We the search the initial page can get lost. Eviction can kick in and delete that key if it is already obsolete.